### ISSUE TITLE
Exception when CC="" is set

### ISSUE THREAD

**Issue by emersion** [mesonbuild/meson#4522]
Meson 0.48.2

Running

```
export CC=""
meson build
```

results in:

```
The Meson build system
Version: 0.48.2
Source dir: /home/simon/src/mrsh
Build dir: /home/simon/src/mrsh/build
Build type: native build
Project name: mrsh
Project version: 0.0.0
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 364, in run
    app.generate()
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 136, in generate
    self._generate(env)
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 153, in _generate
    intr = interpreter.Interpreter(b)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 1936, in __init__
    self.parse_project()
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 365, in parse_project
    self.evaluate_codeblock(self.ast, end=1)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 404, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 398, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 409, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 705, in function_call
    return func(node, posargs, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 133, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 162, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 2555, in func_project
    self.add_languages(proj_langs, True)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 2698, in add_languages
    (comp, cross_comp) = self.detect_compilers(lang, need_cross_compiler)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 2617, in detect_compilers
    comp = self.environment.detect_c_compiler(False)
  File "/usr/lib/python3.7/site-packages/mesonbuild/environment.py", line 620, in detect_c_compiler
    return self._detect_c_or_cpp_compiler('c', 'CC', want_cross)
  File "/usr/lib/python3.7/site-packages/mesonbuild/environment.py", line 509, in _detect_c_or_cpp_compiler
    compilers, ccache, is_cross, exe_wrap = self._get_compilers(lang, evar, want_cross)
  File "/usr/lib/python3.7/site-packages/mesonbuild/environment.py", line 483, in _get_compilers
    if compilers[0] == 'ccache':
IndexError: list index out of range
```

**Comment by eli-schwartz** [comment]
As of today, this seems to ignore the variable entirely.

```
$ CC="" meson setup builddir-emptycc/
The Meson build system
Version: 0.60.1
Source dir: /home/eschwartz/git/mesonbuild/muon
Build dir: /home/eschwartz/git/mesonbuild/muon/builddir-emptycc
Build type: native build
Project name: muon
Project version: 0.0.1
C compiler for the host machine: ccache cc (clang 13.0.0 "clang version 13.0.0")
C linker for the host machine: cc ld.bfd 2.36.1
[...]
```

**Comment by eli-schwartz** [comment]
This issue got duplicated by #5451 and fixed in #5601
